core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1252,7 +1252,7 @@ class DAGScheduler(
                 }
               }
 
-              submitWaitingStages()
+              submitWaitingChildStages(shuffleStage)
```

Comment:
Should this be done when `!shuffleStage.isAvailable` and we have resubmitted the `shuffleStage`, or only within the `else` branch?